### DISCUSSION TITLE
Tailwind CLI performance?

### DISCUSSION THREAD

**Discussion by SebastianStehle** [saadeghi/daisyui#2839] [Q&A] [answered]
Hi,

i have a very small setup with daisy UI, tailwind CLI and a small css file with around 100 custom lines.

Whenever I make changes to this file the tailwind CLI takes around 5 to 15sec to recompile. Very annoying. And even worse, I get a lot of OOM errors and crashes during the day.

Have you made a similar experience? I am not sure if this is due to the additional complexity of daisy UI, I cannot find similar issues in the tailwind repo.

**Comment by NikarashiHatsu** [comment]
Navigating this matter involves considering various factors, with the chief culprit often being the speed of your computing system. Remarkably, on Apple M1, this predicament is nearly unheard of, as it effortlessly compiles the CSS file within a matter of seconds.

To delve deeper into troubleshooting, it would be beneficial to provide details about your system, enabling a more thorough investigation of the issue at hand.

**Reply by SebastianStehle** [reply]
Good point. It is a Windows 11 machine, with

* `i9-14900KF`
* `64 GB RAM
* `Samsung SSD 980 PRO 2`

So I am pretty sure, that this is not the problem. I also made the content part of the CLI as specific as possible:

```
module.exports = {
  content: [
    "./Pages/**/*.cshtml*",
    "./wwwroot/**/*.html*",
    "./wwwroot/**/*.js*"
  ],
  ```
  
  There are not that much files in these folders. Like 20 max.

**Reply by NikarashiHatsu** [reply]
That's a spectacle of computer you have. It's really surprising to see Tailwind CSS acting strangely in such a high-powered environment.

I hadn't considered that the problem might be related to a memory leak or your computer being under stress. However, have you tried compiling the CSS without using DaisyUI to see if that improves performance? It could be worth a shot to see if that makes a difference.

**Reply by SebastianStehle** [reply, marked as answer]
Yeah, I have tried that, but it makes almost no difference. 

But I have solved the problem. I have figured out that I can easily measure the performance with the --inspect argument.

So this is my result:
![image](https://github.com/saadeghi/daisyui/assets/1236435/b0917190-8900-468a-a0b1-b84c48b2e927)

Using this tool I can also debug the CLI and I realized that the CLI was scanning files that I have not seen before. So I found a forgotten node_modules folder that was there because the project has been restructured a while ago and it was scanning all these files. The folder was hidden by the IDE. Stupid me. Now it runs in 60ms. Not super impressive, but more than enough :)

Here is the profile, in case you are interested in my stupidity :D

[CPU-20240210T234618.cpuprofile](https://github.com/saadeghi/daisyui/files/14230746/CPU-20240210T234618.cpuprofile)

Lets see if my memory leaking issues are also gone with that.
